app/controllers/events_controller.rb

```diff
@@ -85,6 +86,11 @@ def set_event
       @event = Event.find(params[:id])
     end
 
+    # Get the type of event that should be created
+    def event_type
+      params[:type] || Event
```

Comment:
I'd remove the `|| Event`. Creating an abstract event will most likely lead to issues further down the road. Otherwise this solution seems pretty damn sweet. I was trying to figure out how to make it clean for multiple hours and ended up just adding case-when statements because I couldn't figure out how to it properly.